internal/testing/e2e/in_memory.go

```diff
@@ -101,79 +88,153 @@ func (iom *inmemoryOM) HealthCheck() error {
 }
 
 func (iom *inmemoryOM) Context() context.Context {
-	return iom.mainTc.Context()
-}
-
-func (iom *inmemoryOM) cleanup() {
-	iom.mc.Close()
-	iom.mainTc.Close()
-	iom.mmfTc.Close()
-	iom.evalTc.Close()
+	return context.Background()
 }
 
 func (iom *inmemoryOM) cleanupMain() error {
 	return nil
 }
 
-// Create a minimatch test service with function bindings from frontendService, backendService, and queryService.
-// Instruct this service to start and connect to a fake storage service.
-func createMinimatchForTest(t *testing.T, evalTc *rpcTesting.TestContext) *rpcTesting.TestContext {
-	var closer func()
+func newInMemoryEnvironment(t *testing.T) config.View {
+
+	mredis := miniredis.NewMiniRedis()
```

Comment:
I would really like to avoid templating where reasonable.

Also in the future I want to read this in and compare it against the real cluster's config map.  In that case I'd also have to define a bunch of values for that in two places.  The mental model for me here is "we're starting with what the real cluster tests uses, then we're being explicit about what places we're cheating."